### PR TITLE
Prevent trash_ids from going to DB

### PR DESCRIPTION
**Pull request by Avatar690**
Prevent trash ids from going to database (they are TRASH, why do you want them?!)  Tested over the last 2 months on my setup.  Does not interfere with any other DB insertion.  Updates pokemon count as normal.  Greatly reduces database overhead (in my case, we are not encountering 21 of the most common pokemon.  Results in an 80% decrease in database rows).

### CODE REVIEW

**Comment by evilmoses** [comment]
Nice, but I would say that it should be a config setting so people can choose whether to enforce it or not

**Comment by bbdoc** [comment]
Agree with evilmoses... should be a config setting. I used to have a lot of pokes in trash_ids just because I wanted them hidden on map by default and was quite confused after you updated that on your fork not to see them anymore on my map ;-) Took some time to figure out. So should be a setting with default value to False

**Comment by wateredstorm** [comment]
This would be pretty dopped VS murkrows pidgeys and all that stuff that has no medal or anysort of duty for mapping :P

**Comment by Avatar690** [comment]
Can easily drop a config option in.

File: monocle/db_proc.py [hunk 1]
```diff
@@ -40,10 +40,13 @@ def run(self):
                 item_type = item['type']
 
                 if item_type == 'pokemon':
-                    db.add_sighting(session, item)
-                    self.count += 1
+                    if item['pokemon_id'] in conf.TRASH_IDS:
+                         self.count += 1
```
Comment (MonkeyTedBaker) [inline monocle/db_proc.py]: This indeed, makes sense. 
Why 5 spaces for tab though?

**Comment by MonkeyTedBaker** [comment]
really guys? TRASH_IDS is not the same as IGNORE_SAVE_POKEMON_IDS?

File: monocle/db_proc.py (hunk 1 above)
Comment (Avatar690) [reply monocle/db_proc.py]: Got an extra space in there?  Easily removed.  Will update later.

**Comment by Avatar690** [comment]
Updated

**Comment by XGrendelX** [comment]
Traceback (most recent call last):
  File "scan.py", line 28, in <module>
    from monocle.worker import Worker
  File "/home/map/newhaven/Monocle/monocle/worker.py", line 15, in <module>
    from .db import FORT_CACHE, MYSTERY_CACHE, SIGHTING_CACHE, RAID_CACHE
  File "/home/map/newhaven/Monocle/monocle/db.py", line 12, in <module>
    from . import bounds, spawns, db_proc, sanitized as conf
  File "/home/map/newhaven/Monocle/monocle/db_proc.py", line 44
    if item['pokemon_id'] in conf.NO_DB_INSERT_IDS:
                                                  ^
TabError: inconsistent use of tabs and spaces in indentation

**Comment by Avatar690** [comment]
Really.... really?!  Stupid IDLE.  Ill fix.

File: monocle/db_proc.py [hunk 2]
```diff
@@ -40,8 +40,15 @@ def run(self):
                 item_type = item['type']
 
                 if item_type == 'pokemon':
-                    db.add_sighting(session, item)
-                    self.count += 1
+                    if conf.NO_DB_INSERT_IDS:
```
Comment (MonkeyTedBaker) [inline monocle/db_proc.py]: Thanks for the PR. Sorry for the late reply. I think the check should be done in worker.py
https://github.com/MonkeyTedBaker/Monocle/blob/develop/monocle/worker.py#L820

If we do it in db_proc.py, we would waste some resources such as encounter, notifications, etc before deciding to insert or not. I think it is important to move it.

**Comment by Avatar690** [comment]
Would require refactoring the db_proc.  Pokemon sightings are sent to the db processor and parsed for sightings and spawnpoints.  I tested it previously upstream in worker.py and found that spawnpoints weren't going to the db on bootstrap

**Comment by phonezawphyo** [comment]
I see, get the problem during bootstrap. Perhaps, we should check if spawnpoint for sighting is already found before applying the filter, all in worker.py. If sp is not yet found, save the sighting without applying the filter. Should be doable?

**Comment by MonkeyTedBaker** [comment]
I see, get the problem during bootstrap. Perhaps, we should check if spawnpoint for sighting is already found before applying the filter, all in worker.py. If sp is not yet found, save the sighting without applying the filter. Should be doable?

**Comment by XGrendelX** [comment]
Thanks for making a PR addressing my request. Enjoying the testing of it on my maps! I had something similar I was doing through mirroring my db's but it is nice to just have this built into the native scanner.

**Comment by Avatar690** [comment]
@MonkeyTedBaker that would require an io to the disk to check the pickle and more logic.  Also wouldn't help on new spawnpoints added as it might take forever to finally discover a new point.  I think modding the db proc as is is a plus gain on the front end.  Backend probably evens out because you aren't inserting as many.  I noticed a neglible decrease in util when I started truly dumping trash_ids (I also wasn't considering a none type... Which I considered here).  All in all, it is probably a push on the backend.  On the front, it helps having your db 60-80% of the previous size (bonus, queries are more efficient).  Just my 2cents.  Can certainly look at refactoring long term.

**Comment by Avatar690** [comment]
@XGrendelX np.  I had done it already and tested. Easy pr.

**Comment by Avatar690** [comment]
@XGrendelX np.  I had done it already and tested. Easy pr.

**Pull request merged**